src/main/java/tdl/record_upload/RecordAndUploadApp.java

```diff
@@ -68,18 +72,54 @@ public static void main(String[] args)  {
             // Prepare remote destination
             AWSSecretProperties awsSecretProperties = AWSSecretProperties
                     .fromPlainTextFile(Paths.get(params.configFile));
-            Destination s3BucketDestination =  S3BucketDestination.builder()
-                    .awsClient(awsSecretProperties.createClient())
-                    .bucket(awsSecretProperties.getS3Bucket())
-                    .prefix(awsSecretProperties.getS3Prefix())
-                    .build();
+
+            boolean syncFolder = !params.doNotSync;
+            Destination uploadDestination;
+            if (syncFolder) {
+                uploadDestination = S3BucketDestination.builder()
+                        .awsClient(awsSecretProperties.createClient())
+                        .bucket(awsSecretProperties.getS3Bucket())
+                        .prefix(awsSecretProperties.getS3Prefix())
+                        .build();
+            } else {
+                uploadDestination = new NoWorkDummyDestination();
+            }
+
 
             // Validate destination
             log.info("Checking permissions");
-            s3BucketDestination.testUploadPermissions();
+            uploadDestination.testUploadPermissions();
+
+            // Timestamp
```

Comment:
These comments say they can be descriptive method names so the steps are encapsulated, clear and visible